plugins/action/tests/integration/ndfc_network_validate.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+from pprint import pprint
+import json
+from ansible.utils.display import Display
+from ansible.plugins.action import ActionBase
+from ansible.module_utils.six import raise_from
+from ansible.errors import AnsibleError
+from ansible.module_utils.common.text.converters import to_native
+from ..plugin_utils.tools import load_yaml_file, process_deepdiff
+from ..plugin_utils.pydantic_schemas.dcnm_network.schemas import DcnmNetworkQuerySchema
+
+try:
+    from deepdiff import DeepDiff
+except ImportError as imp_exc:
+    DEEPDIFF_IMPORT_ERROR = imp_exc
+else:
+    DEEPDIFF_IMPORT_ERROR = None
+
+if DEEPDIFF_IMPORT_ERROR:
+    raise_from(
+        AnsibleError('DeepDiff must be installed to use this plugin. Use pip or install test-requirements.'),
+        DEEPDIFF_IMPORT_ERROR)
+
+display = Display()
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        results = super(ActionModule, self).run(tmp, task_vars)
+        results['failed'] = False
+
+        if 'ndfc_data' not in self._task.args:
+            results['failed'] = True
+            results['msg'] = 'No data found in ndfc_data'
+            return results
+
+        ndfc_data = self._task.args['ndfc_data']
+        # removes ansible embeddings and converts to native python types
+        native_ndfc_data = json.loads(json.dumps(ndfc_data, default=to_native))
+
+        if "test_data" not in self._task.args:
+            results['failed'] = True
+            results['msg'] = 'No test data is found in test_data'
+            return results
+
+        test_data = self._task.args['test_data']
+
+        if "config_path" not in self._task.args:
+            results['failed'] = True
+            results['msg'] = 'No path is loaded into config_path'
+            return results
+
+        config_path = self._task.args['config_path']
+
+        if 'check_deleted' in self._task.args:
+            check_deleted = self._task.args['check_deleted']
+
+        check_deleted = False
+
+        test_fabric = test_data['test_fabric']
+        if config_path != "":
+            # only parse if config file exists
+            expected_config_data = load_yaml_file(config_path)
+            expected_data = DcnmNetworkQuerySchema.yaml_config_to_dict(expected_config_data, test_fabric)
+
+            expected_data_parsed = DcnmNetworkQuerySchema.parse_obj(expected_data).dict(exclude_none=True)
+
+        ndfc_data_parsed = DcnmNetworkQuerySchema.parse_obj(native_ndfc_data).dict(exclude_none=True)
+
+        if check_deleted:
```

Comment:
For a cleaner top level test flow I think it would be good to move the code under `if check_deleted` into it's own helper function inside this file and then just call it here.

```python
def verify_deleted(check_deleted, existing_networks, ndfc_parsed):
    if not check_deleted:
        return

    # ... check_deleted code here... #
```